zilliqa/benches/it.rs

```diff
@@ -308,20 +465,29 @@ fn a_big_process_vote(big: &mut Consensus, vote: Vote) -> Proposal {
     });
     assert_eq!(
         proposal.transactions.len(),
-        4000,
-        "proposal {} is not full",
+        txns_per_block,
+        "proposal {} is not the expected size",
         proposal.view()
     );
+
     proposal
 }
 
 fn b_tiny_process_block(tiny: &mut Consensus, from: PeerId, proposal: Proposal) {
+    let last_txn = proposal.transactions.last().map(|t| t.calculate_hash());
     let (_, tiny_vote) = tiny
         .proposal(from, black_box(proposal), false)
         .unwrap()
         .unwrap();
     // We assert 'tiny' actually voted but don't do anything with its vote.
     assert!(matches!(tiny_vote, ExternalMessage::Vote(_)));
+
+    // Get the last transaction receipt and make sure it succeeded. We assume that all other transactions succeeded if
+    // this one did.
+    if let Some(last_txn) = last_txn {
```

Comment:
:+1:

Reply:
Why do we not need to confirm this in `c_big_process_block` also?

Reply:
They process the same block and I didn't want to run this check more than necessary - It isn't what we actually want to benchmark, just a sanity check that our benchmark is working.